Fix issue with nav bar component

Mariamkhaled45 opened the pull request.

Commit:
Created Navbar

Commit:
refactor: fix nav bar in mobile view

change nav bar display from horizontal to vertical view after menu icon is pressed in mobile view

Fixes issue #15

Ahmed-Khaled24 commented on src/components/NavBar.jsx:
```diff
@@ -0,0 +1,68 @@
+          </div>
+        </div>
+
+        <div className="nav__toggle" id="nav-toggle" onClick={toggleMenu}>
+          <IoMenu />
+        </div>
```
We need to use this icon instead of the current 

![nav-menu](https://github.com/ASU-Semicolon/The-New-Dashboard/assets/80486237/231d6539-888a-402e-b034-d68f223eed13)
you can download it and replace the current one

Ahmed-Khaled24 commented on src/components/NavBar.css:
![image](https://github.com/ASU-Semicolon/The-New-Dashboard/assets/80486237/453ce95b-a898-4200-9e67-f37c4893cf25)
Spacing between links here is very tight we need to make it a bit larger.
Also, it is shifted to the left we need it to be in the middle.
**hint**: use display flex not block

Commit:
refactor: fix nav bar to dynamically display list items and CSS adjustments
nav bar list items display fixed to be more dynamic
necessary CSS adjustments done for improved styling and layout consistency
